src/services/ghost/GhostStrategy.ts

```diff
@@ -0,0 +1,181 @@
+import * as vscode from "vscode"
+import { parsePatch } from "diff"
+
+export class GhostStrategy {
```

Comment:
I like the strategy structure with prompt + parsing together. would be nice to util-ify this